File: BLL/Models/WorkingProgramDetailsModel.cs
using DAL.Entities;

namespace BLL.Models
{
    public class WorkingProgramDetailsModel : IDomainModel
    {
        public Guid Id { get; set; }

        public string Name { get; set; }

        public IdNameModel<Guid> Subject { get; set; }

        public IdNameModel<Guid> EducationalProgram { get; set; }
        
        public string CreatedByName { get; set; }

        public string ApprovedByName { get; set; }

        public bool IsAvailable { get; set; }

        public IEnumerable<CommentModel> Comments { get; set; }
    }
}

File: WebApi/ClientApp/src/app/models/working-program-details.model.ts
import { Guid } from "guid-typescript";
import { IdNameModel } from "./id-name-model.model";

export interface WorkingProgramDetails {
    id: Guid
    name: string
    subject: IdNameModel
    educationalProgram: IdNameModel
    createdByName: string
    approvedByName: string
    isAvailable: boolean
}
